consensus/types/src/participation_flags.rs

```diff
@@ -0,0 +1,87 @@
+use crate::{consts::altair::NUM_FLAG_INDICES, test_utils::TestRandom, Hash256};
+use safe_arith::{ArithError, SafeArith};
+use serde_derive::{Deserialize, Serialize};
+use ssz::{Decode, DecodeError, Encode};
+use test_random_derive::TestRandom;
+use tree_hash::{TreeHash, TreeHashType};
+
+#[derive(Debug, Default, Clone, Copy, PartialEq, Deserialize, Serialize, TestRandom)]
+#[serde(transparent)]
+#[cfg_attr(feature = "arbitrary-fuzz", derive(arbitrary::Arbitrary))]
+pub struct ParticipationFlags {
+    bits: u8,
+}
+
+impl ParticipationFlags {
+    pub fn add_flag(&mut self, flag_index: u32) -> Result<(), ArithError> {
+        if flag_index > NUM_FLAG_INDICES as u32 {
```

Comment:
Just noticed this should be a `>=`. Fixing it in the alpha.5/6 branch, but can back port it as necessary.